Builder.cpp

```diff
@@ -0,0 +1,119 @@
+#include <iostream>
+#include <string>
+
+class ProcessorType
+{
+    public:
+        std::string processor;
+};
+
+class Sensor
+{
+    public:
+        float Msensor;
+};
+
+
+class Camera
+{
+    public:
+        ProcessorType*  processorType;
+        Sensor* sensor;
+    
+        void specifications()
+        {
+            std::cout << "Sensor: " << sensor->Msensor << " megapixel "<< std::endl;
+            std::cout << "Processor Type: " << processorType->processor << std::endl;
+        }
+};
+
+class Builder
+{
+    public:
+        virtual ProcessorType* getProcessorType() = 0;
+        virtual Sensor* getSensor() = 0;
+};
+
+class Director
+{
+    Builder* builder;
+
+    public:
+        void setBuilder(Builder* newBuilder)
+        {
+            builder = newBuilder;
+        }
+
+        Camera* getCamera()
+        {
+            Camera* camera = new Camera();
+
+            camera->sensor = builder->getSensor();
+
+            camera->processorType = builder->getProcessorType();
+
+            return camera;
+        }
+};
+
+class CanonBuilder : public Builder
```

Comment:
With this implementation it is not `CanonBuilder` it is more "Canon specifications builder".

Let's discuss it internally in the next class.